### DISCUSSION TITLE
strain-based plasticity models

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#17666] [Q&A Modules: Solid mechanics] [answered]
Hi everyone,

I'm considering implementing some strain-based plasticity models, and i'm wondering whether anyone has done this in MOOSE already, or even toyed with the idea.

My reason for doing this is to more easily/accurately capture post-yield behaviour for rocks.  When rocks fail (Mohr-Coulomb plasticity is often used) and further deform, their mechanics is quite different from the pre-yield elastic behaviour.  Eg, consider pulling apart a rock via a tensile stress: after failure the rock has fractured so essentially has "zero Young's modulus" (this may not be really true, because tension in an orthogonal direction will encounter competent rock), but then upon recompaction to its original state, the Young's modulus will assume its original value, since the fracture will have "squashed back" to nothing.  This mechanics may be captured with softening and hardening of the plastic strengths, but that ends up being computationally quite challenging.

So, i'm toying with strain-based plasticity.   Any comments?

Example article:

https://doi.org/10.1016/S0045-7825(96)01245-5

**Comment by hugary1995** [comment, marked as answer]
So first possibly dumb question I have: what you described is pretty much just softening plasticity. How does "strain-based" factor in here?

**Reply by hugary1995** [reply]
So after reading the abstract of that article, it sounds like strain-based and stress-based plasticity are equally valid choices, but one may be more convenient than the other in certain cases. Correct me if I'm wrong.

So is your focus here "strain space plasticity" or "softening plasticity"? Even if you want both in the end, it may be easier to pursue one of them first, than the other one.

I don't know how hard "strain-space plasticity", but I guess all it needs is a different set of material models in moose. Maybe the return mapping is done on strain?

"Softening plasticity" is trickier, as you may need fixed point iteration to obtain a solution post yielding.

**Reply by WilkAndy** [reply]
You're right - i believe that strain-based *might* be more convenient in my case.  This is just an intuition at the moment, as i haven't tried anything.

I'm not really interested in softening.  I use it a lot, and find it's too computationally expensive, even if it appears to capture reality somewhat realistically in some cases.

I'm not sure whether i'm going to pursue this strain-based-plasticity work.  At the moment i'm just wondering if anyone has done stuff in MOOSE (or otherwise) using it.
